cpp/tests/test_populations.cpp

```diff
@@ -224,3 +225,29 @@ TEST(TestPopulations, set_difference_from_group_total)
         }
     }
 }
+
+TEST(TestPopulations, populations_constraints)
+{
+    // Verify that the functions check_constraints() and apply_constraints() work as expected.
+    mio::Populations<double, InfectionState, AgeGroup> pop(
+        {mio::Index<InfectionState>(InfectionState::Count), mio::Index<AgeGroup>(7)});
```

Comment:
This should work as well.
```suggestion
    mio::Populations<double, InfectionState, AgeGroup> pop({InfectionState::Count, AgeGroup(7)});
```

Reply:
Actually, i get an error if i use this. At least for Agegroup, i need the explicit definition of mio:: Index, for the InfectionState your suggestion is enough...

Reply:
That is weird, an AgeGroup *is* an Index\<AgeGroup\> by definition. And it works for e.g. the osecir model.

Reply:
Actually this is not true as AgeGroup is defined seperately in this test and does not inherit from Index<AgeGroup. Thats also why we need this here!

Reply:
I did not see that. That makes a lot more sense...
We can emphasize that AgeGroup and InfectionState are not defined in mio to avoid this confusion. One option is to add a leading `::` to them to indicate the global namespace, alternatively (or additionally) you could rename them to e.g. TestAgeGroup/TestInfectionState.

Reply:
I like the idea with the renaming.